### PR TITLE
Add the function of reading sonar data and return the returned intensities from .JSF file

### PR DESCRIPTION
**Pull request by xyp8023**

### CODE REVIEW

File: src/data_tools/src/jsf_data.cpp [hunk 1]
```diff
@@ -11,21 +11,271 @@
...
+
+    if (!pings.empty()) {
+        pings[0].first_in_file_ = true;
+    }
+    return pings;
+
+}
+
+
+template <typename ReturnType>
```
Comment (nilsbore) [inline src/data_tools/src/jsf_data.cpp]: You do not need a template here as far as I can tell, please remove it.

Comment (nilsbore) [reply src/data_tools/src/jsf_data.cpp]: Or if you want to make it generic over different kinds of types, use it in the return type and inside the function. But then you would need to make `read_jsf_file` generic as well.

**Comment by nilsbore** [comment]
Ok, apart from the nit it looks good. Let me know when you've resolved it and I'll merge it into master.

**Comment by xyp8023** [comment]
I've resolved it.

File: src/data_tools/src/jsf_data.cpp [hunk 2]
```diff
@@ -270,10 +269,9 @@ jsf_sss_ping::PingsT parse_file_impl(const boost::filesystem::path& path){
 namespace std_data{
 using namespace jsf_data;
 
-template <>
-jsf_sss_ping::PingsT parse_file<jsf_sss_ping>(const boost::filesystem::path& file)
+jsf_sss_ping::PingsT parse_file(const boost::filesystem::path& file)
```
Comment (nilsbore) [inline src/data_tools/src/jsf_data.cpp]: There should still have been a `template <>` block before *this* function.

File: src/data_tools/src/test_jsf.cpp [hunk 3]
```diff
@@ -24,12 +24,12 @@ using namespace std_data;
 
 int main(int argc, char** argv){
     int rtn=0;
-    // boost::filesystem::path path("/home/yipingxie/Downloads/SSH-0007-KTH1-20190615-090603.JSF");
+    boost::filesystem::path path("/home/yipingxie/Downloads/SSH-0007-KTH1-20190615-090603.JSF");
```
Comment (nilsbore) [inline src/data_tools/src/test_jsf.cpp]: Maybe use argv[1] instead of a hardcoded string?

File: src/data_tools/src/test_jsf.cpp [hunk 4]
```diff
@@ -24,12 +24,12 @@ using namespace std_data;
...
     // boost::filesystem::path path("/home/yipingxie/Downloads/SSL-0037-low1-20190615-111128.JSF");
-    boost::filesystem::path path("/home/yipingxie/Downloads/SSH-0023-high9-20190615-102808.JSF");
+    // boost::filesystem::path path("/home/yipingxie/Downloads/SSH-0023-high9-20190615-102808.JSF");
 
 
-    jsf_sss_ping::PingsT pings = parse_file<jsf_sss_ping>(path);
+    jsf_sss_ping::PingsT pings = parse_file(path);
```
Comment (nilsbore) [inline src/data_tools/src/test_jsf.cpp]: Again, this should have the template.

File: src/data_tools/include/data_tools/jsf_data.h [hunk 5]
```diff
@@ -62,7 +62,6 @@ void show_waterfall_image(const jsf_sss_ping::PingsT& pings);
 
 namespace std_data {
 
-template <>
 jsf_data::jsf_sss_ping::PingsT parse_file(const boost::filesystem::path& file);
```
Comment (nilsbore) [inline src/data_tools/include/data_tools/jsf_data.h]: There should be an empty template clause here.

**Comment by nilsbore** [comment]
You removed the template in the wrong place :).  It should have been the `template <typename ReturnType>` where I added the first comment.

**Comment by xyp8023** [comment]
Aha! I did not pay much attention to the differences before. Now I fix it.

**Comment by nilsbore** [comment]
Thank you, now it looks good.

**Pull request merged**